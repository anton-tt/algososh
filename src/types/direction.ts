export enum Direction {
  Ascending = "ascending",
  Descending = "descending",
  Empty = ""
}
