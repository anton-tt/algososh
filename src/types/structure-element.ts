import { ElementStates } from "./element-states";

export type TDataStructureElement = {
  value: string;
  state: ElementStates;
};

export type TNumberArrayElement = {
  value: number;
  state: ElementStates;
};