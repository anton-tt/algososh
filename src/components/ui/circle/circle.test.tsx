import { render } from "@testing-library/react";
import { Circle } from "./circle";
import { ElementStates } from "../../../types/element-states";

describe("Проверка компонента Circle", function() {
  
  it("отрисовка без буквы", () => {
    const { container } = render(<Circle letter="" />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с буквой", () => {
    const { container } = render(<Circle letter="A" />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с head", () => {
    const { container } = render(<Circle head="head" />)
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с react-элементом в head", () => {
    const { container } = render(<Circle head={<Circle isSmall={true} />} />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с tail", () => {
    const { container } = render(<Circle tail="tail" />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с react-элементом в tail", () => {
    const { container } = render(<Circle tail={<Circle isSmall={true} />} />);
    expect(container).toMatchSnapshot();
  });
  
  it("отрисовка с index", () => {
    const { container } = render(<Circle index={45} />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка с пропом isSmall ===  true", () => {
    const { container } = render(<Circle isSmall={true} />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка в состоянии default", () => {
    const { container } = render(<Circle state={ElementStates.Default} />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка в состоянии changing", () => {
    const { container } = render(<Circle state={ElementStates.Changing} />);
    expect(container).toMatchSnapshot();
  });

  it("отрисовка в состоянии modified", () => {
    const { container } = render(<Circle state={ElementStates.Modified} />);
    expect(container).toMatchSnapshot();
  });
  
}); 