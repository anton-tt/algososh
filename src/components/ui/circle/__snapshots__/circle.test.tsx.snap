// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Проверка компонента Circle отрисовка без буквы 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка в состоянии changing 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   changing"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка в состоянии default 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка в состоянии modified 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   modified"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head string"
    >
      head
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с index 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    >
      45
    </p>
    <div
      class="text text_type_input text_color_input mt-4 absolute tail60 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с react-элементом в head 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с react-элементом в tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    >
      <div
        class="content "
      >
        <div
          class="text text_type_input text_color_input mb-4 absolute head element"
        />
        <div
          class="circle  small default"
        >
          <p
            class="text text_type_circle text_color_input letter"
          />
        </div>
        <p
          class="text text_type_input text_color_input mt-4 absolute index"
        />
        <div
          class="text text_type_input text_color_input mt-4 absolute tail30 element"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с tail 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 string"
    >
      tail
    </div>
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с буквой 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle   default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      >
        A
      </p>
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;

exports[`Проверка компонента Circle отрисовка с пропом isSmall ===  true 1`] = `
<div>
  <div
    class="content "
  >
    <div
      class="text text_type_input text_color_input mb-4 absolute head element"
    />
    <div
      class="circle  small default"
    >
      <p
        class="text text_type_circle text_color_input letter"
      />
    </div>
    <p
      class="text text_type_input text_color_input mt-4 absolute index"
    />
    <div
      class="text text_type_input text_color_input mt-4 absolute tail30 element"
    />
  </div>
</div>
`;
